server/src/main/java/org/opensearch/index/mapper/TextFieldMapper.java

```diff
@@ -608,9 +615,13 @@ public ValueFetcher valueFetcher(QueryShardContext context, SearchLookup searchL
             return SourceValueFetcher.toString(name(), context, format);
         }
 
-        void setAnalyzer(NamedAnalyzer delegate) {
+        void setAnalyzer(String name, NamedAnalyzer delegate) {
```

Comment:
This is `NamedAnalyzer` right, why do we need to pass name if we could do `delegate.name()` ?

Reply:
Yeah, `name` can be get by `delegate.name()` directly, I've changed the code, thanks!